test/websocket-reconnect.js

```diff
@@ -0,0 +1,40 @@
+import WebSocket from 'ws'
+import test from 'ava'
+import openWebSocket from '../src/open-websocket'
+
+const TEST_SERVER_PORT = 8080
+
+test.beforeEach(t => {
+  const server = new WebSocket.Server({ port: TEST_SERVER_PORT })
+  server.on('connection', ws => {
+    ws.on('message', () => server.clients.forEach(client => client.send('test')))
+  })
+
+  t.context.server = server
+})
+
+test.afterEach.always(t => {
+  return new Promise(resolve => {
+    t.context.server.close(() => {
+      resolve()
+    })
+  })
+})
+
+test.serial('[WS] reconnect', t => {
```

Comment:
I would simply rather create the server in the same serial test you are doing, and close it there too

Reply:
But what happens if the test fails? Then there is no guarantee that the server is being closed?